www/__tests__/input-matcher.test.ts

```diff
@@ -0,0 +1,249 @@
+import { 
+    TlEntry, 
+    UserInputForTrip, 
+    fmtTs, 
+    printUserInput, 
+    validUserInputForDraftTrip, 
+    validUserInputForTimelineEntry,
+    getNotDeletedCandidates,
+    getUserInputForTrip,
+    getAdditionsForTimelineEntry,
+    getUniqueEntries
+} from '../js/survey/input-matcher';
+
+describe('input-matcher', () => {
+    let userTrip: UserInputForTrip;
+    let trip: TlEntry;
+
+    beforeEach(() => {
+        // create valid userTrip and trip object before each test case.
+        userTrip  = { 
+            data: {
+                end_ts: 1437604764,
+                start_ts: 1437601247,
+                label: 'FOO',
+                status: 'ACTIVE'
+            },
```

Comment:
These tests look great! Could you add comments in the code explaining
- where the fake data comes from - are they based on real trips? Are they from `real_examples` on the server? Are they made-up values?
- how this testing strategy parallels the testing on the server (I think you referred to `TestUserInputFakeData.py` while making these?)

This way, anyone who updates these tests in the future will know where all of this came from.

Reply:
Updated the comment. Please check this commit: b8ec27d